### PR TITLE
Integrated build

### PR DESCRIPTION
**Pull request by chedburgh**
This is quite a large refactor. It creates a new hdbpp_cm version2, that is compatible only with the refactored devicer servers + backends. 

Highlights:

* New CMake build system
* Clang integration
* Refactoring and code modernization to get rid of some clang warnings.
* Removed libhdbpp dependency. Add attribute done by Event Subscriber.
* Updated README for new build system
* Observe new namespace in libhdbpp

### CODE REVIEW

**Comment by chedburgh** [comment]
I will be merging in the fixes from the master to here shortly.

File: README.md [hunk 1]
```diff
@@ -1,76 +1,83 @@
...
 Tango device server for the HDB++ Configuration Manager
 
-## Version
-
-The current release version is 1.0.0
+## Documentation
 
-### **Important Changes** 0.9.0 -> 1.0.0
+* See the Tango documentation [here](http://tango-controls.readthedocs.io/en/latest/administration/services/hdbpp/index.html#hdb-an-archiving-historian-service) for broader information about the HB++ archiving system and its integration into Tango Controls
+* hdbpp-es [CHANGELOG.md](https://github.com/tango-controls-hdbpp/libhdbpp/blob/master/CHANGELOG.md) contains the latest changes both released and in development.
```
Comment (lorenzopivetta) [inline README.md:10]: Shouldn't this read hdbpp-cm ?

**Review by lorenzopivetta — APPROVED** [review]
I've not tested running the device, but the build env works.
Looks good to me.

**Review by gscalam — COMMENTED** [review]
Since we're doing some cleanings it could be the occasion to remove some useless code. For sure we can remove everything compiled only if _USE_FERMI_DB_RW is defined. I think also code within _MULTI_TANGO_HOST define is useless, it was a request by ESRF, but at the end never used I think, @bourtemb can you confirm?

**Comment by bourtemb** [comment]
> 
> 
> Since we're doing some cleanings it could be the occasion to remove some useless code. For sure we can remove everything compiled only if _USE_FERMI_DB_RW is defined. I think also code within _MULTI_TANGO_HOST define is useless, it was a request by ESRF, but at the end never used I think, @bourtemb can you confirm?

Yes, I confirm we don't use this MULTI_TANGO_HOST feature on our HDB++ subscribers and configurator.
Still I would do that additional cleanup in separate Pull Request(s).
In general, I really prefer small Pull Requests, easier to review and focusing on only 1 topic.

**Review by dlacoste-esrf — APPROVED** [review]

**Review by bourtemb — CHANGES_REQUESTED** [review]
This PR cannot be merged automatically because it has some conflicts on  src/HdbConfigurationManager.cpp which need to be solved first.
A rebase would be welcome so we don't lose the memory leak fix on src/HdbConfigurationManager.cpp

File: README.md (hunk 1 above)
Comment (dlacoste-esrf) [reply README.md:10]: Indeed, did it on master after merging.

**Pull request merged**